mono/metadata/class.c

```diff
@@ -5808,6 +5817,7 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 	if (klass->generic_container) {
 		klass->is_generic = 1;
 		klass->generic_container->owner.klass = klass;
+		klass->generic_container->is_anonymous = FALSE; // Owner class is now known, container is no longer anonymous
```

Comment:
Now it makes sense. Yeah, it would be nice if the container state is not changed after it's created.